### ISSUE TITLE
Locked cells and undo/redo

### ISSUE THREAD

**Issue by RMacfarlane** [microsoft/vscode#99162]
Issue Type: <b>Bug</b>

Testing https://github.com/microsoft/vscode/issues/98991

1. Create github-issues notebook with two cells
2. Type in a cell, then lock it

Try to undo, nothing happens. So my expectation is then that locked cells will not respond to undo. 

3. Type in the other cell
4. Repeatedly undo

I'm able to remove the text from the locked cell.

VS Code version: Code - Insiders 1.46.0-insider (1bfa086adb9271ff56be11821da686e4cfffb672, 2020-06-02T08:10:00.819Z)
OS version: Darwin x64 18.7.0

**Comment by rebornix** [comment]
Good catch. Will look into this next week.

**Comment by rebornix** [comment]
The concept of **Lock** is pretty challenging here as what it does now is making the editor readonly, so users can't modify the content through the editor widget. However, the content can still be updated by Undo/Redo (as the undo stack is shared), F2 (through workspace edits), file changes from local disk, etc. I think it's true for regular file as well. cc @jrieken

**Comment by jrieken** [comment]
Yeah, we have no real concept of locking, just toggling readonlyness.

**Comment by rebornix** [comment]
@jrieken *Toggle readonlyness* is much better fit.
